packages/web-react/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createElement } from 'react';
+import { createPortal } from 'react-dom';
+import { SpiritModalProps } from '../../types';
+import { useClassNamePrefix } from '../../hooks';
+import ModalBackdrop from './ModalBackdrop';
+import ModalContent from './ModalContent';
+import ModalDialog from './ModalDialog';
+import ModalCloseButton from './ModalCloseButton';
+import { useModal } from './useModal';
+
+const Modal = (props: SpiritModalProps) => {
+  const {
+    id,
+    children,
+    closeOnBackdrop = true,
+    closeOnEscape = true,
+    contentProps,
+    backdropProps,
+    dialogProps,
+    closeButtonProps,
+    parentSelector = '#root',
+    showBodyClose = true,
+    UNSAFE_style,
```

Comment:
Please, move logic with style and class handling into `useModalStyleProps`. For example look at `useAlertStyleProps.` All components must be composable.